app/lib/getNearbyServices.js

```diff
@@ -1,27 +1,53 @@
 const log = require('../../app/lib/logger');
-const docDB = require('../../config/config').docDB;
-const queries = require('../lib/queries');
-const filterServices = require('../lib/filterServices');
-
-function getNearbyServices(req, searchPoint, limits, next) {
-  const geoSpatialQuery = queries.geoSpatialQuery(searchPoint, limits.searchRadius);
-
-  req.app.locals.docDBClient
-    .queryDocuments(docDB.collectionUrl, geoSpatialQuery)
-    .toArray((err, results) => {
-      if (err) {
-        log.error({ err }, 'DocumentDB query failed');
-        next(err);
-      } else {
-        log.debug(`${results.length} RESULTS returned...`);
-
-        log.debug('filter-and-sort-services-start');
-        const filteredServices = filterServices(results, limits);
-        log.debug('filter-and-sort-services-end');
+const MongoClient = require('mongodb').MongoClient;
+const co = require('co');
+const mongodbConfig = require('../../config/config').mongodb;
+const filterServices = require('./filterServices');
+const VError = require('verror').VError;
 
-        next(null, filteredServices);
+const connectionString = mongodbConfig.connectionString;
+
+function getNearbyServices(searchPoint, limits, next) {
+  co(function* a() {
+    const db = yield MongoClient.connect(connectionString);
+    log.debug(`Connected to ${mongodbConfig.connectionString}`);
+
+    const col = db.collection(mongodbConfig.collection);
+
+    col.aggregate([{
+      $geoNear: {
+        near: { type: 'Point', coordinates: searchPoint.coordinates },
+        distanceField: 'dist',
```

Comment:
Changing the distanceField here and subsequently everywhere  breaks the sorting. api-test_1               |   32 passing (942ms)
api-test_1               |   1 failing
api-test_1               | 
api-test_1               |   1) filterServices happy path should return the nearest obj first:
api-test_1               | 
api-test_1               |       AssertionError: expected 'A' to equal 'B'
api-test_1               |       + expected - actual
api-test_1               | 
api-test_1               |       -A
api-test_1               |       +B
api-test_1               |       
api-test_1               |       at Context.it (test/unit/lib/filterServices.js:76:50)

I believe the aggregate results are past to the filtering but the sorting is also changing the data so the result give me an outlier (which is why I went with a different var to begin with, IIRC)
![screenshot-localhost 3000 2017-02-10 08-38-29](https://cloud.githubusercontent.com/assets/1902266/22819946/83394d98-ef6c-11e6-9100-3f5d28277490.png)

Changing to `dist` and raising an issue to investigate further if needed.